### PR TITLE
Added QuietDrift support for Curtain 3

### PR DESCRIPTION
**Pull request by den-mko**
Added speed to open, close and set_position methods. "00" or "ff" - normal, "01" - slow #222

### CODE REVIEW

File: switchbot/devices/curtain.py [hunk 1]
```diff
@@ -63,27 +68,31 @@ async def _send_multiple_commands(self, keys: list[str]) -> bool:
         return final_result
 
     @update_after_operation
-    async def open(self) -> bool:
-        """Send open command."""
-        return await self._send_multiple_commands(OPEN_KEYS)
+    async def open(self, speed="ff") -> bool:
```
Comment (bdraco) [inline switchbot/devices/curtain.py:71]: Please change speed to be an int and convert it in the function to hex

File: switchbot/devices/curtain.py [hunk 2]
```diff
@@ -63,27 +68,31 @@ async def _send_multiple_commands(self, keys: list[str]) -> bool:
...
 
     @update_after_operation
     async def stop(self) -> bool:
         """Send stop command to device."""
         return await self._send_multiple_commands(STOP_KEYS)
 
     @update_after_operation
-    async def set_position(self, position: int) -> bool:
-        """Send position command (0-100) to device."""
+    async def set_position(self, position: int, speed="ff") -> bool:
```
Comment (bdraco) [inline switchbot/devices/curtain.py:90]: Please add a test for these changes

**Review by bdraco — CHANGES_REQUESTED** [review]

File: switchbot/devices/curtain.py (hunk 2 above)
Comment (den-mko) [reply switchbot/devices/curtain.py]: There are tests for parsers but not for functions like open/close. Which tests do you mean? The only thing i made was replacing ff flag with speed flag.

*Sorry it's a bit tough for me - i am not a python dev so i struggle a bit

**Comment by bdraco** [comment]
I adjusted the code a bit to make it easier to follow.

Please retest. I'll merge it and do a release if it works as expected

**Comment by den-mko** [comment]
Small typing fix. Tested successfully.

**Review by bdraco — APPROVED** [review]

**Comment by Brephlas** [comment]
There is still a pull request needed to upgrade the integration in the HA core similar to [this](https://github.com/home-assistant/core/pull/101164) one, right? 
@den-mko Do you want to create one? If not, I could do that too

**Comment by den-mko** [comment]
I already wrote the code with curtains speed changing features but did not have time to submit a PR. Will try to submit it tomorrow

**Comment by lightheaded** [comment]
It has now been incorporated into HA core. How would one go about using quietdrift now?

**Comment by quenthal** [comment]
> I already wrote the code with curtains speed changing features but did not have time to submit a PR. Will try to submit it tomorrow

Is it already in your fork?

**Comment by quizzical86** [comment]
@den-mko would you happen to have any news on enabling support for quietdrift?

**Pull request merged**